src/content/post/20161129_git_reflog_amend.md

```diff
@@ -0,0 +1,114 @@
++++
+date        = "2016-11-29T08:51:09+02:00"
+title       = "Trapped by an amend, saved by a reflog"
+tags        = ['git']
+topics      = ['git']
+description = "Personal experience on getting in trouble with a common git command and finding my way out."
+slug        = 'trapped-by-amend-saved-by-reflog'
+author      = "Marcello"
++++
+[SparkFabrik]: http://www.sparkfabrik.com/  "SparkFabrik"
+
+We use Git on a daily basis, and most of the time we always enter the same commands.
+It can happen to found ourselves trapped by one of them, sometimes because we forgot a side-effect, other times because we confused the right option.
+That's OK, because within Git most of the time lies the appropriate and easiest solution. So, let me tell you how I was hit by an `--amend` during a commit and how I easily got out.
+
+<!--more-->
+
+Sometimes during the day, especially in the time right before lunch or going home, it's possible to forget one the most important rules of the savvy developer: in the terminal, you can be quick with all the keys, but think twice before pressing "enter".
+
+Breaking this rule is what happened to me some days ago, but that was luckily a chance to remind me how much powerful git is and how sometimes the solutions to an apparently bad situation can be just a few commands away.
+
+## The amend secret
+
+As many of you probably know, the `git commit --amend` command can be used to edit the message of the last commit in the history. This is one of the most common uses because - as developers - we are prone to typos and bad phrasing.
+But the `--amend` option can also be used to include *new code* in the last commit.
+
+For example.
+Let's say you've just created a commit but you've also forgot to add a block comment to that very method you've written, and you don't want to have another commit in the history, for various reasons. You have many choices at this point, including creating a new commit with the new lines and squash it with the previous one using the `git rebase` command.
+
+But you can also do this:
+
+<pre>
+# Add new line and prepare them for the commit.
+git add -p
+# Amend the previous commit and add to it the staged modifications.
+git commit --amend
+</pre>
+
+A this point, you are asked to edit the previous commit message and the staged edits will be included in it. Useful, isnt'it?
+
+## The side effect
+
+This little trick can be very handy in many situations, but it can also be a very sharp double-edge blade because, as you probably have guessed, you can use the amend on *any* commit including, for example, a *merge commit*.
+
+So, here's a possible sequence of commands that could generate a bad situation.
+
+<pre>
+# You add some modifications.
+git add -p
+# Then create a new commit
+git commit -m 'Added new public method'
+# Then merge the last develop in your working branch
+git fetch --all
+git merge origin/develop
+</pre>
+
+So far the history of your branch could be something like:
+
+<pre>
```

Comment:
Just noticed you are using `<pre>` tags where a `\`\`\`` can do. I don't know if this impacts the syntax highlight... can you please check?

Thanks